platform/platform-filter-clientinfo/src/main/java/org/codice/ddf/platform/filter/clientinfo/ClientInfoInjector.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.filter.clientinfo;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+/**
+ * This CXF filter extracts client-specific information and places it in a {@link ThreadLocal}
+ * instance so it can be forwarded to the {@code CatalogFramework} as an {@link AccessPlugin}.
+ */
+public class ClientInfoInjector implements Filter, AccessPlugin {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientInfoInjector.class);
+
+    private static final String CLIENT_INFO_KEY = "client-info";
+
+    private final ThreadLocal<ClientInfoContainer> clientInfo;
+
+    public ClientInfoInjector() {
+        this.clientInfo = ThreadLocal.withInitial(new Supplier<ClientInfoContainer>() {
+            @Override
+            public ClientInfoContainer get() {
+                return null;
+            }
+        });
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {}
+
+    @Override
+    public void destroy() {}
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+            FilterChain filterChain) throws IOException, ServletException {
+        ClientInfoContainer infoContainer = new ClientInfoContainer(servletRequest);
+        clientInfo.set(infoContainer);
+        filterChain.doFilter(servletRequest, servletResponse);
```

Comment:
I would suggest wrapping this in a try-finally and in the finally block you can remove the client info.  You can use `ThreadContext` from Shiro that will do this for you but then this filter will have to run after `LoginFilter`.